core/src/main/java/org/zenoss/zep/index/impl/BaseEventIndexBackend.java

```diff
@@ -34,12 +38,44 @@
 
     private final Map<String, SS> savedSearches = new ConcurrentHashMap<String, SS>();
 
+    private final LoadingCache<EventFilter, EventTagSeveritiesSet> tagSeveritiesCache;
+
+    private static final Date IMMEDIATELY = new Date(0);
     private static final Logger logger = LoggerFactory.getLogger(EventIndexBackend.class);
 
-    protected BaseEventIndexBackend(Messages messages, TaskScheduler scheduler, UUIDGenerator uuidGenerator) {
+    protected BaseEventIndexBackend(Messages messages, TaskScheduler scheduler, UUIDGenerator uuidGenerator,
+                                    int tagSeverityCacheSize, int tagSeveritiesCacheTTL) {
         this.messages = messages;
         this.scheduler = scheduler;
         this.uuidGenerator = uuidGenerator;
+        this.tagSeveritiesCache = CacheBuilder.
+                newBuilder().
+                maximumSize(tagSeverityCacheSize).
+                refreshAfterWrite(1, TimeUnit.SECONDS).
```

Comment:
Per the docs ( http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/cache/CacheBuilder.html#refreshAfterWrite%28long,%20java.util.concurrent.TimeUnit%29 ), it seems like every entry in the cache will be refreshed unnecessarily after any operation on the cache. Is that the case? This seems like a lot of extra load, even though it's asynchronous.